src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy

```diff
@@ -1092,6 +1092,34 @@ public enum DarknessAblaze implements LogicCardInfo {
           attackRequirement {}
           onAttack {
             damage 130
+            afterDamage {
+              def waterE = self.cards.energyCount W
+              if (waterE) {
+                def energies = []
+                def count = min waterE, 2
+                def finalCount = 0
+                count.times {
+                  def info = "Select [W] Energy to return to your hand."
+                  def energy = self.cards.findAll {energyFilter W}.select(info)
+                  if (energy instanceof specialEnergy) {
+                    def types = energy.getEnergyTypesOverride()
+                    types.each {
+                      if (it.contains W) {
+                        finalCount += 1
+                      }
+                    }
+                  }
+                  else {
+                    finalCount += 1
+                  }
+                  energies.add(energy)
+                  if (finalCount >= count) return
```

Comment:
Asking from ignorance: is return a way to break this loop? In case this doesn't build, maybe something along the lines of:

```
def energyRange = 1..count
for (n in energyRange) {
	//whatever
  if (finalCount >= count) break
}
```

Reply:
Return exits a closure. Break will break out of the switch statement.

Reply:
Actually, break might work fine in a normal loop like this. It's other closures that it would break out of the switch statement I think. Guess we'll see how it reacts.